x/ibc/light-clients/solomachine/types/proof.go

```diff
@@ -0,0 +1,217 @@
+package types
```

Comment:
I should have thought about this earlier, but I wonder if we should make these all proto structs instead of this kinda-arbitrary bytestring concatenation encoding? cc @warner any thoughts?

Reply:
This can be deferred to a follow-up, though, since this file is nicely abstracted (thanks)